rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -268,12 +287,59 @@ func (r *Layer2Relayer) commitGenesisBatch(batchHash string, batchHeader []byte,
 // ProcessPendingBatches processes the pending batches by sending commitBatch transactions to layer 1.
 func (r *Layer2Relayer) ProcessPendingBatches() {
 	// get pending batches from database in ascending order by their index.
-	dbBatches, err := r.batchOrm.GetFailedAndPendingBatches(r.ctx, r.cfg.BatchSubmission.MaxBatches)
+	allBatches, err := r.batchOrm.GetFailedAndPendingBatches(r.ctx, 0)
 	if err != nil {
 		log.Error("Failed to fetch pending L2 batches", "err", err)
 		return
 	}
 
+	// if backlog outgrow max size, force‐submit enough oldest batches
+	backlogCount := len(allBatches)
+	backlogMax := 75 //r.cfg.BatchSubmission.BacklogMax
+
+	if len(allBatches) < r.cfg.BatchSubmission.MinBatches || len(allBatches) == 0 {
+		log.Debug("Not enough pending batches to submit", "count", len(allBatches), "minBatches", r.cfg.BatchSubmission.MinBatches, "maxBatches", r.cfg.BatchSubmission.MaxBatches)
+		return
+	}
+
+	// return if not hitting target price
+	if backlogCount <= backlogMax {
```

Comment:
somehow this code block is very ugly. Maybe you can put the logic into a function and then just return with the error message that you want to log. In that case we can continue and submit immediately (and log the error message). this way we can use the Go typical guard style